### PR TITLE
feat/AF-187: Create install command

### PR DESCRIPTION
**Pull request by LucasR29**
# Install Command Implementation

This branch introduces the `install` command for the CLI tool. Key features include:

- Setup of the installation process for Anyflow in a user's project
- Integration with Hardhat configuration
- Handling of environment variables for project setup
- User prompts for necessary configuration details
- Error handling and validation for the installation process

The main goal is to provide a good and function installation experience for users setting up Anyflow.

**TESTING**

```
pnpm install
```
```
pnpm build
```
```
pnpm anyflow install
```

### CODE REVIEW

**Comment by LucasR29** [comment]
The commit ```[fix: conflicts with ts after hardhat inclusion](https://github.com/AnyFlowLabs/anyflow-cli/pull/2/commits/5537ee221b1dbc3d24647f896fa10565faa8c259)``` fixes a problem with TypeScript that has existed since I started developing the CLI.

Basically, the tsconfig file had a wrong configuration. I patched this problem when I started this branch, but when using Hardhat became necessary, the solution I found was not working anymore. With these changes, I believe that the problem is fixed once and for all.

File: hardhat.config.ts
Comment (yudikubota) [inline hardhat.config.ts:1]: I see you get the right idea of importing a subset of our default hardhat.config file. But I still think it can be a little simpler and less cluttlered. For example, this is what we have in the backend after we merge the config files:

```
import originalHardhatConfig from './original.hardhat.config'

const overwriteConfig = { ... }

export default { ...originalHardhatConfig, ...overwriteConfig };
```

What you could do in the user's project is something similar, but allow the user to customize their config. Something like:

```
import { mergeConfig } from '@anyflow/cli'

const userConfig = { ... }

export default mergeConfig(userConfig, { ...additionalOptionsIfNeeded });
```

This provides a much cleaner UX, customization and setup for the user and allow us to properly merge configs. Please note that we do need to merge the configs deeply and not just overwrite them.

https://lodash.com/docs/4.17.15#merge

---

Additionally, as this project's source is public I would suggest creating a comment on those PKs to make it clear we only use them as proxies and they do not impose any risk being public. I've made a commit including this comment.

File: config/internal-config.ts
Comment (yudikubota) [inline config/internal-config.ts:2]: it seems both `RPC_BASE_URL` and `CHAIN_DEPLOYMENT_ID` are currently constant. 

`CHAIN_DEPLOYMENT_ID` will come from the deployment step so we can hold it for now and implement it later but `RPC_BASE_URL` should be changeable through ENV to allow us for local testing and change environments.

File: src/commands/auth/store-token/store.ts
Comment (yudikubota) [inline src/commands/auth/store-token/store.ts:23]: From a UX perspective, I would prefer to leave it to use the fallback if keytar fails instead of guiding the user to read a docs. There's no problem in using local file.

File: hardhat.config.ts
Comment (yudikubota) [reply hardhat.config.ts:1]: PS. I see with the updated code this was resolved. Ignore the message above.

Comment (yudikubota) [inline hardhat.config.ts:1]: When I tested it with a JS file it has resulted in the following config:
```
import AnyflowHardhatConfig from "anyflow-cli/hardhat.config";
require("@nomicfoundation/hardhat-toolbox-viem");
require("@nomicfoundation/hardhat-ignition-viem");

const config = {
  solidity: "0.8.24",
};

module.exports = config;
```

Please note .js do not use import.

**Review by yudikubota — CHANGES_REQUESTED** [review]
Some optional comments. But this still need work with .js files.

**Review by yudikubota — APPROVED** [review]

**Pull request merged**
